lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -4062,6 +4062,46 @@ def stem(self, x, y, z, *, linefmt='C0-', markerfmt='C0o', basefmt='C3-',
 
     stem3D = stem
 
+    @_preprocess_data()
+    def arrow3d(self, end, start=None, **kwargs):
+        """
+        3D plot of a single arrow
+
+        Parameters
+        ----------
+        end : 1D array
+            an array of shape (3,).
+
+        start : 1D array, default: (0,0,0)
+            an array of shape (3,).
+
+        **kwargs
+            All other keyword arguments are passed on to
+            `~mpl_toolkits.mplot3d.art3d.Arrow3D`.
+
+        Returns
+        -------
+        arrow : `~mpl_toolkits.mplot3d.art3d.Arrow3D`
+
+        """
+        had_data = self.has_data()
+        if start is None:
+            start = np.zeros_like(end)
+        if np.shape(end) != (3,):
+            raise ValueError("end must be an array of length 3")
+        if np.shape(start) != (3,):
+            raise ValueError("start must be an array of length 3")
+        # Set default arrow properties and update with any additional keyword args
+        arrow_props = dict(
+            mutation_scale=20, arrowstyle="-|>", shrinkA=0, shrinkB=0
+        )
+        arrow_props.update(kwargs)
+        xs, ys, zs = [start[0], end[0]], [start[1], end[1]], [start[2], end[2]]
+        arrow = art3d.Arrow3D(xs, ys, zs, **arrow_props)
+        self.add_artist(arrow)
+        self.auto_scale_xyz(xs, ys, zs, had_data)
+        return arrow
```

Comment:
Add line breaks between different code blocks to make it easier to distinguish between.

```suggestion
        had_data = self.has_data()

        if start is None:
            start = np.zeros_like(end)
        if np.shape(end) != (3,):
            raise ValueError("end must be an array of length 3")
        if np.shape(start) != (3,):
            raise ValueError("start must be an array of length 3")

        # Set default arrow properties and update with any additional keyword args
        arrow_props = dict(
            mutation_scale=20, arrowstyle="-|>", shrinkA=0, shrinkB=0
        )
        arrow_props.update(kwargs)

        xs, ys, zs = [start[0], end[0]], [start[1], end[1]], [start[2], end[2]]
        arrow = art3d.Arrow3D(xs, ys, zs, **arrow_props)
        self.add_artist(arrow)
        self.auto_scale_xyz(xs, ys, zs, had_data)

        return arrow
```